### DISCUSSION TITLE
Inquiry: Multi-UE Emulation Without ZMQ-Based RF Devices

### DISCUSSION THREAD

**Discussion by rajugbv** [srsran/srsRAN_Project#531] [Ideas & Suggestions]
Dear Community,

I hope this message finds you well. I recently came across a tutorial on Multi-UE Emulation, demonstrating the connection of multiple UEs to a single gNB utilizing ZMQ-based RF devices. This approach has sparked my interest in exploring alternative methodologies.

Would anyone be able to advise if there's a feasible method to achieve multi-UE connectivity to a single gNB without relying on ZMQ-based RF devices? I am keen on understanding different configurations or technologies that could facilitate this setup, potentially enhancing our current project's scope and capabilities.

Thank you in advance for your insights and guidance. I look forward to any suggestions or experiences you might share regarding this matter.

Best regards,
Raju

**Comment by pgawlowicz** [comment]
Hi, you can always connect multiple UEs to a gnb over the air using USRPs.

**Reply by rajugbv** [reply]
Quick question:  have Already added the new SIM details in open5GS for UE2. Do I need to tweak anything else in the UEs or gNB configs for this?

**Reply by pgawlowicz** [reply]
In case of srsUE you need to change usim section in config.

**Reply by rajugbv** [reply]
Do I need to adjust anything in the gNB configuration file to accommodate this setup, and should I activate the following parameters in the UE configuration file?
[gw]
netns = ue2
ip_devname = tun_srsue
ip_netmask = 255.255.255.0
Additionally, is creating a namespace for ue2 necessary?"

**Reply by pgawlowicz** [reply]
yes, you need to create a network namespace for each emulated UE as described in our tutorial : https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html#id19

No changes are required on the gnb side.
